tests/test_training.py

```diff
@@ -294,6 +294,7 @@ def setUp(self):
 
     def test_early_stopper_best_epoch_model_retrieval(self):
         """Test if the best epoch model is returned when using the early stopper."""
+        # TODO this test is failing
```

Comment:
@mberr this test is failing

Reply:
since this is a higher level test, it would be good for us to make a smaller unit test with fewer components in it that causes the same incorrect behavior